test/Conversion/TorchToMhlo/reduction.mlir

```diff
@@ -241,3 +241,64 @@ func.func @torch.aten.max(%arg0: !torch.vtensor<[?,?,?],f32>) -> !torch.vtensor<
   %0 = torch.aten.max %arg0 : !torch.vtensor<[?,?,?],f32> -> !torch.vtensor<[],f32>
   return %0 : !torch.vtensor<[],f32>
 }
+
+// -----
+// CHECK-LABEL:   func.func @torch.aten.frobenius_norm.dim(
```

Comment:
I would recommend not adding any .mlir tests. I don't think they add much in this case. Also see our testing philosophy here:

https://github.com/llvm/torch-mlir/blob/main/docs/architecture.md#dos-and-donts-for-unit-vs-end-to-end-testing

> DON'T use a unit test if your lowering pattern could be described as a trivial "macro expansion" of one op into another op or set of ops. That is, if you feel like your unit test is just rewriting b.create<...>(...) into CHECK: ... then it is probably not a useful unit test.

Reply:
The mlir is for testing conversion for new op from torch to mhlo. If you don't like it, I will withdraw the changes.

Reply:
We should not submit this until ReduceFrobeniusNormModule_basic and ReduceFrobeniusNormKeepDimModule_basic tests pass. Once those tests pass, I think this .mlir test is irrelevant.

Why do the new E2E tests fail e2e with TorchToMhlo?

Reply:
> Why do the new E2E tests fail e2e with TorchToMhlo?

That's because `hlo-legalize-to-linalg` pass will invoke element-wise ops' (like mhlo.sqrt) reifyReturnTypeShapes function in case that the result type has dynamic shape. See codes [here](https://github.com/tensorflow/mlir-hlo/blob/master/lib/Dialect/mhlo/transforms/legalize_to_linalg_utils.cc#L76-L78). This will introduce ShapeOfOp, which will not be converted or eliminated in the following passes and finally cause an error.